### ISSUE TITLE
Conflict with spell check

### ISSUE THREAD

**Issue by PierreMarchand20** [yzhang-gh/vscode-markdown#475]
#### What is the problem?

When I write math, [spell check](https://github.com/streetsidesoftware/vscode-spell-checker) raises warnings for math commands:
![image](https://user-images.githubusercontent.com/15056323/59701203-5a544d00-91f5-11e9-9cc4-1feaa17bef8a.png)

I know it is possible to avoid that, because they do it in [LaTeX-Workshop](https://github.com/James-Yu/LaTeX-Workshop), would it be possible to do it also with this extension ? 

I have no idea how it works :-/

#### How can I reproduce it?
Use [spell check](https://github.com/streetsidesoftware/vscode-spell-checker) and use math commands in latex environments.

**Comment by yzhang-gh** [comment]
Thanks.

I guess it is the spell checker that has made some adaptions to the LaTeX syntax. To my knowledge, we cannot do anything here. (The spell checker just takes in the text and returns some warnings)

**Comment by PierreMarchand20** [comment]
I looked at it quickly, the spell checker I talked about (it is the most used in VS Code, more than 3 millions download) has some configuration tools to ignore parts of a document. Here's how they did it in the latex extension : https://github.com/James-Yu/LaTeX-Workshop/pull/78 (seems quite easy)

I do not have time to do it (writing a phd thesis...), but I will do it later if you cannot find the time to do it.

**Comment by PierreMarchand20** [comment]
it is quite a powerful extension, it can be configured to spell checks comments in a code, so of course you can make it _not_ spell check the code ^^

**Comment by yzhang-gh** [comment]
Thanks for the information.

It looks like they finally came to some _user-level_ configurations? (The PR was deleted and had never been merged...)

For Markdown math, those configs can be `\\$.+?\\$` and `\\$\\$[\w\W]+?\\$\\$`. (Simply suppress spell checking in the math environment 😅)

I didn't see some _extension-level_ configurations (for this extension). (Didn't take a closer look. Please let me know if I am wrong.)

**Comment by PierreMarchand20** [comment]
You are totally right, actually it looks like cSpell just has a big dictionary with all the Tex commands, and it is enabled for Tex files.

But I looked at how we can embed a language in another one in vscode (so latex in markdown here), may be we can use what is described here https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#embedded-languages

**Comment by yzhang-gh** [comment]
I am afraid it is not a solution as we expected. As it says,

> This ensures that bracket matching, commenting, and other basic language features work as expected in the embedded language.

It may not help with the spell check. (On the other hand, KaTeX is different from LaTeX)

---

For now, I would suggest using the above RegExps first. (I don't think there is too much to check inside the math environments)

I will try to find a good way later.

**Comment by PierreMarchand20** [comment]
I thought that may be if we tell vscode that it is another language may be it would call the right extension... vscode really lacks a feature like “language injection” or “embedded language”, and it would be really convenient for markdown. It would be nice to use snippets and IntelliSense from other extensions for latex and other languages instead of redefining everything here, there is actually an old issue about that : https://github.com/microsoft/vscode/issues/1751

In the mean time, I agree that the regexpr above will solve my issue. But since it works out of the box for latex, may be we can ask the maintainer of cSpell to do something for markdown, I will post an issue.

**Comment by yzhang-gh** [comment]
Cannot agree more
